### ISSUE TITLE
COBOL - Conditional compilation and EXEC SQL INCLUDE SQLCA

### ISSUE THREAD

**Issue by FALLAI-Denis** [IBM/zopeneditor-about#379]
## Development environment used

- Z Open Editor version: 3.3.3
- Editor Platform
  - [X] Visual Studio Code
  - [ ] Red Hat CodeReady Workspaces
  - [ ] Eclipse Che
  - [ ] Standalone Theia
- Editor Platform Version: 1.85.2
- Operating System: Windows 10 22H2 Enterprise x64
- Java Version: 17
- [ ] Related to RSE API?
  - RSE API Plugin version:
  - Zowe CLI version:
  - Node.js version:
- Logs attached: no

## Problem Description

Almost all of the source code analysis issues related to the use of conditional compilation have been resolved in the latest versions of ZOE, and once again I thank you for your efforts.

However, a new case has just been identified:
* EXEC SQL INCLUDE SQLCA directive is not handled correctly when under conditional compilation control
* this results in the emission of false problems on the declaration of zones declared by the SQLCA (which is not a real copybook)

### Observed behavior

With SQLCA declaration under conditional compilation (even if the condition is true):
![image](https://github.com/IBM/zopeneditor-about/assets/63861965/0c10b0bf-15e2-49b8-9c36-4f143976be4f)
* Note that the problems are reported twice.

With SQLCA declaration not under conditional compilation:
![image](https://github.com/IBM/zopeneditor-about/assets/63861965/397c5170-ea3d-4d3e-a5ae-40ee0cde5976)

This suggests that the COBOL Language Server processes the EXEC SQL statements before having processed the conditional compilation.

**Comment by Russell-May** [comment]
Can you provide me with a sample program where you are seeing this issue?

**Comment by FALLAI-Denis** [comment]
Hi,

Sample code:

```
       IDENTIFICATION DIVISION.
       PROGRAM-ID. TSTSQLCA.
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       01  HOWMUCH PIC S9(9) COMP-5.
       >>define isDB2 as b'1'
       >>if isDB2
           exec sql include sqlca end-exec.
       >>end-if
       PROCEDURE DIVISION.
       MAIN.
           exec sql
             select count(*) into :HOWMUCH
             from SYSIBM.SYSDUMM1
           end-exec
           if SQLCODE not = 0 and SQLCODE not = +100
             display "Oups !"
           end-if
           .
       BYE.
           goback.
```

![image](https://github.com/IBM/zopeneditor-about/assets/63861965/49167393-e5bf-453b-8184-eafa019d6894)

![image](https://github.com/IBM/zopeneditor-about/assets/63861965/1409cbdd-6922-455f-ba0f-87020aa005ab)

**Comment by Russell-May** [comment]
Thanks, I see the issue and will see what can be done to fix it.

**Comment by phaumer** [comment]
Fixed in v3.4.0.
